pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -180,14 +185,9 @@ def on_advance_end(self):
         # update plateau LR scheduler after metrics are logged
         self.update_lr_schedulers("step", update_plateau_schedulers=True)
 
-        self.total_batch_idx += 1
-
         # progress global step according to grads progress
         self._increment_accumulated_grad_global_step()
 
-        if self.done:
-            raise StopIteration
-
```

Comment:
@awaelchli Do you know why we had this?

This is `on_advance_end` so the next thing we do is check the while loop condition

Reply:
there must have been other code below these lines that since got deleted, and there was a specific reason (tests) that would need to avoid running these lines before the next done() check.